Tests/MapboxNavigationTests/NavigationMapViewTests.swift

```diff
@@ -356,6 +356,68 @@ class NavigationMapViewTests: TestCase {
         }
     }
     
+    func testGenerateRouteLineGradientWithSingleCongestion() {
+        let coordinates: [CLLocationCoordinate2D] = [
+            CLLocationCoordinate2D(latitude: 37.798, longitude: -122.398),
+            CLLocationCoordinate2D(latitude: 37.795, longitude: -122.398),
+            CLLocationCoordinate2D(latitude: 37.795, longitude: -122.395),
+        ]
+        let congestionSegment: CongestionSegment = (coordinates, CongestionLevel.low)
+        var feature = Feature(geometry: .lineString(LineString(congestionSegment.0)))
+        feature.properties = [
+            CongestionAttribute: String(describing: congestionSegment.1),
+            CurrentLegAttribute: true
+        ]
+        let congestionFeatures:[Turf.Feature] = [feature]
+        
+        var fractionTraveled = 0.0
+        var routeLineGradient = navigationMapView.routeLineGradient(congestionFeatures, fractionTraveled: fractionTraveled, isMain: true, isSoft: false)
+        XCTAssertEqual(routeLineGradient[0.0], navigationMapView.trafficLowColor)
+        XCTAssertEqual(routeLineGradient[1.0], navigationMapView.trafficLowColor)
+        
+        fractionTraveled = 0.3
+        var fractionTraveledNextDown = Double(CGFloat(fractionTraveled).nextDown)
+        routeLineGradient = navigationMapView.routeLineGradient(congestionFeatures, fractionTraveled: fractionTraveled, isMain: true, isSoft: false)
+        XCTAssertEqual(routeLineGradient[0.0], navigationMapView.traversedRouteColor)
+        XCTAssertEqual(routeLineGradient[1.0], navigationMapView.trafficLowColor)
+        XCTAssertEqual(routeLineGradient[fractionTraveled], navigationMapView.trafficLowColor)
+        XCTAssertEqual(routeLineGradient[fractionTraveledNextDown], navigationMapView.traversedRouteColor)
+        
+        fractionTraveled = 0.999999999
+        fractionTraveledNextDown = Double(CGFloat(fractionTraveled).nextDown)
+        routeLineGradient = navigationMapView.routeLineGradient(congestionFeatures, fractionTraveled: fractionTraveled, isMain: true, isSoft: true)
```

Comment:
Can you please add verification for cases when `GradientCongestionFadingDistance` was modified and for fraction, which is exactly at the gradient itself, not when fraction is already has plain color.

Reply:
Thanks for the suggestion, it would be included and solved in the tail work in this ticket #3365